### ISSUE TITLE
Scoping

### ISSUE THREAD

**Issue by ShahneRodgers** [ShahneRodgers/keep-in-touch#1]
* What features do we want?
* What clients (web-only, or mobile too)?
* What languages/frameworks would we prefer?

**Comment by ShahneRodgers** [comment]
For features, I was thinking:

1. do we want a user to sign up and be given a guid-based link to other people interested in keeping in touch, where the user ends up being "responsible" for actually doing the keeping in touch (ie they're the only one notified), or do we want it to be more social-network-like where both people have to have an account and then the frequency of reminders is dictated by an average of what both people choose as their preferred frequency?
2. should we let people specify the ideal medium of keeping in touch (eg email, txt, facebook, messenger, twitter, instagram, github, snailmail, etc)?
3. how random do we want it to be? Should the probability of being reminded max at, say, 80% and always be >= 0.5% or something? Should that be customizable?
4. What details should a user be able to store about someone - I think fields and reminders for birthdays, work place, significant others would be useful and maybe free-form notes about the last conversation.
5. Do we want to try to generate an interesting discussion prompt or something silly like that?

**Comment by dewitpj** [comment]
Surprising, I think mobile will be used more - thinking of things like Facebook etc - pretty sure they have more app users than web users ?

**Comment by ShahneRodgers** [comment]
Yeah, for sure. But mobile is a pain since you have to get it onto the app stores (and pay for that).

**Comment by dewitpj** [comment]
> Yeah, for sure. But mobile is a pain since you have to get it onto the app stores (and pay for that).

Web will work for the start :)

**Comment by ShahneRodgers** [comment]
We could go with a progressive web app. I think those can be downloaded on phones too.

**Comment by dewitpj** [comment]
> 1. do we want a user to sign up and be given a guid-based link to other people interested in keeping in touch

I like this alot - we can group it by "our user" and workplace - I know have had different groups of people from the same work place - some might just be a "linkedin" type group, others might be an active facebook type group

> 2. should we let people specify the ideal medium of keeping in touch (eg email, txt, facebook, messenger, twitter, instagram, github, snailmail, etc)?

A per "group" (ref above) setting - "Group A" I want an email, but "Group B" send me a WA

> 3. how random do we want it to be? Should the probability of being reminded max at, say, 80% and always be >= 0.5% or something? Should that be customizable?

What if we also base this around "times" - when you left the company, started, Christmas, birthdays etc ?

> 4. What details should a user be able to store about someone - I think fields and reminders for birthdays, work place, significant others would be useful and maybe free-form notes about the last conversation.

Sounds good

> 5. Do we want to try to generate an interesting discussion prompt or something silly like that?

At some point, yeah ?

**Comment by ShahneRodgers** [comment]
> I like this a lot - we can group it by "our user" and workplace - I know have had different groups of people from the same work place - some might just be a "linkedin" type group, others might be an active facebook type group

So you're thinking it'll be for the logged in user to decide rather than a choice for the person signing up? Like say I've got an account; I was thinking people who want me to keep in touch with them could choose how they'd like me to contact them and how frequently. But I think you're thinking that I'd be able to classify groups of people (maybe "close friends", "relatives", "company A"...) and choose how / how frequently to get in touch with them.

I guess another thought would be say we've both got an account, and we both sign up to each others. Do we do anything smart there about alternating responsibilities, or do we just expect each person to double their expected time?
